lib/Doctrine/DBAL/Driver/SQLSrv/SQLSrvStatement.php

```diff
@@ -150,9 +150,13 @@ public function bindParam($column, &$variable, $type = null, $length = null)
      */
     public function closeCursor()
     {
-        if ($this->stmt) {
```

Comment:
@deeky666 with your fix `testReuseStatementAfterClosingCursor()` passes, thank you. Not sure if this is relevant to this issue (I wish it wasn't), but `testReuseStatementWithParameterBoundByReference()` above still doesn't pass on SQL Server and IBM DB2. It's made up and I don't know if it should pass in the first place (however it does on other DBs).

The difference between the two is that the first binds param by value, and the second binds them by reference.